### PR TITLE
CAMEL-10531 Introduce fluent builder for Transformer

### PR DESCRIPTION
**Pull request by igarashitm**

### CODE REVIEW

File: camel-core/src/main/java/org/apache/camel/builder/RouteBuilder.java [hunk 1]
```diff
@@ -53,6 +53,7 @@
     private AtomicBoolean initialized = new AtomicBoolean(false);
     private RestsDefinition restCollection = new RestsDefinition();
     private Map<String, RestConfigurationDefinition> restConfigurations;
+    private List<TransformerBuilder> transformerBuilders = new ArrayList<TransformerBuilder>();
```
Comment (apupier) [inline camel-core/src/main/java/org/apache/camel/builder/RouteBuilder.java]: Optional: no need to repeat the TransformerBuilder in new ArrayList<TransformerBuilder>();

File: camel-core/src/main/java/org/apache/camel/builder/TransformerBuilder.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+     *
+     * @param scheme scheme name
+     */
+    public TransformerBuilder scheme(String scheme) {
+        this.scheme = scheme;
+        return this;
+    }
+
+    /**
+     * Set the 'from' data type .
```
Comment (apupier) [inline camel-core/src/main/java/org/apache/camel/builder/TransformerBuilder.java]: cab you precise what is expected as "data type"? is it a classname? is a value taken from somewhere elese?

File: camel-core/src/main/java/org/apache/camel/builder/TransformerBuilder.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        return this;
+    }
+
+    /**
+     * Set the 'from' data type using Java class.
+     *
+     * @param clazz 'from' Java class
+     */
+    public TransformerBuilder from(Class<?> from) {
+        this.from = "java:" + from.getName();
```
Comment (apupier) [inline camel-core/src/main/java/org/apache/camel/builder/TransformerBuilder.java]: java: is used in several places, better to extract a constant

File: camel-core/src/main/java/org/apache/camel/builder/TransformerBuilder.java [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        return this;
+    }
+
+    /**
+     * Set the {@code DataFormatDefinition} to be used for the {@code DataFormat} {@code Transformer}.
+     * @see {@code DataFormatTransformerDefinition}, {@code DataFormatTransformer}
+     * 
+     * @param dfd {@code DataFormatDefinition}
+     */
+    public TransformerBuilder withDataFormat(DataFormatDefinition dfd) {
```
Comment (apupier) [inline camel-core/src/main/java/org/apache/camel/builder/TransformerBuilder.java]: optional: what do you think to use a more meaningful parameter name such as dataFormatDefinition, it is longer but easier to read from my point of view

**Comment by davsclaus** [comment]
I think we need to come up with a different name than `from` and `to` because those are so fundamental to routes that people will get confused.

Can you try to come up with other names.

File: camel-core/src/main/java/org/apache/camel/builder/TransformerBuilder.java [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+    }
+
+    /**
+     * Configure a Transformer according to the configurations built on this builder
+     * and register it into given {@code CamelContext}.
+     * 
+     * @param camelContext {@code CamelContext}
+     */
+    public void configure(CamelContext camelContext) {
+        TransformerDefinition transformer;
```
Comment (apupier) [inline camel-core/src/main/java/org/apache/camel/builder/TransformerBuilder.java]: What do you think to extract this big block of if then else in a private method "createTransformerDefinition" ?

**Comment by igarashitm** [comment]
ah good point, thanks - how about just fromType() and toType()?

File: camel-core/src/main/java/org/apache/camel/builder/TransformerBuilder.java (hunk 5 above)
Comment (igarashitm) [inline camel-core/src/main/java/org/apache/camel/builder/TransformerBuilder.java]: Thanks fixed except this one. I prefer not to split as it's short enough and semantically one series.

**Comment by igarashitm** [comment]
OK, changed to fromType() and toType(). Also added some cosmetic changes according to @apupier 's review.

**Comment by igarashitm** [comment]
Sorry hold on a sec, for consistency I'd like to also change the XML DSL to use same words.

**Comment by igarashitm** [comment]
OK done. I believe it's ready to go.

**Comment by davsclaus** [comment]
Thanks for the PR it has been merged. Do you mind closing this?

**Comment by igarashitm** [comment]
Thanks! Closing.

**Pull request closed without merging**